### DISCUSSION TITLE
Sam3 negative exemplar confusion

### DISCUSSION THREAD

**Discussion by Bstrum36** [ultralytics/ultralytics#24128] [General]
I am getting confusing results on trying to apply negative exemplars.
<img width="1506" height="558" alt="combined" src="https://github.com/user-attachments/assets/ba8f30aa-e5a0-416b-aae7-1760eaf63ed8" />
The image shows the source image with the bounding boxes of the exemplars .  

expected the the opposite of what the image shows.  When I run with labels = [1,1}  I get the same results.

Please advise. 

Here is the prediction 👍 
results = predictor(bboxes=bboxes,
                    labels = [0,0],
                    show=False,
                    boxes = False,
                    )

Here is the environment:

Ultralytics 8.4.21  Python-3.12.11 torch-2.9.0.dev20250716+cu129 CUDA:0 (NVIDIA GeForce RTX 3080 Ti, 12288MiB)
Windows 11

And finally here is all the code 👍 

from ultralytics.models.sam import SAM3SemanticPredictor

import os
import cv2
#
bboxes = [[269.0, 145.0, 313.0, 192.0], [370.0, 190.0, 414.0, 237.0]]
img_path = os.path.join("Images", "baggage_claim.jpg")
img_src = cv2.imread(img_path)
if img_src is None:
    raise FileNotFoundError(f"Image not found: {img_path}")
img = img_src.copy()  # draw boxes on a copy for display

for bbox in bboxes:
    x1, y1, x2, y2 = map(int, bbox)
    cv2.rectangle(img, (x1, y1), (x2, y2), (0, 255, 0), 2)  # Draw green rectangle with thickness 2

# Initialize predictor
overrides = dict(conf=0.25, task="segment", mode="predict", model="sam3.pt", half=True,
                 save=True,name = "sam3_bbox_test",project =  "Output_Images")
predictor = SAM3SemanticPredictor(overrides=overrides)

# Set image
predictor.set_image(img_path)

# Provide bounding box examples to segment similar objects
results = predictor(bboxes=bboxes,
                    labels = [0,0],
                    show=False,
                    boxes = False,
                    )
img_out= results[0].plot(boxes = False)
def side_by_side(left, right):
    if left.shape[0] != right.shape[0]:
        scale = left.shape[0] / right.shape[0]
        new_w = int(right.shape[1] * scale)
        right = cv2.resize(right, (new_w, left.shape[0]), interpolation=cv2.INTER_AREA)
    return cv2.hconcat([left, right])

combined = side_by_side(img, img_out)

#cv2.namedWindow("Result", cv2.WND_PROP_AUTOSIZE)
# cv2.setWindowProperty("Result", cv2.WND_PROP_AUTOSIZE, cv2.WINDOW_AUTOSIZE)
cv2.imshow("Combined ", combined)

print("Showing result. Press spacebar to continue...")
cv2.waitKey(5)
while True:
    key = cv2.waitKey()
    if key == 32:  # 32 is the ASCII code for spacebar
        break
     cv2.destroyAllWindows()

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Bstrum36, thank you for the detailed report and code sample — this is very helpful 🚀. It looks like you may be reporting a SAM3 behavior issue with negative exemplars, so this is an automated response to help gather the right debugging details while an Ultralytics engineer also assists soon 🤝

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. Since you’ve already shared code and environment details, please also include:
- The exact expected behavior versus the actual behavior
- Whether the same result happens after upgrading `ultralytics`
- A minimal test case using the smallest possible image/input that still reproduces the issue
- Any relevant console output or warnings

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by Bstrum36** [reply]
I upgraded as shown:
Ultralytics 8.4.33  Python-3.12.11 torch-2.9.0.dev20250716+cu129 CUDA:0 (NVIDIA GeForce RTX 3080 Ti, 12288MiB)

the code above is minimal.
I get the same results.
I expected the result to have masked the enire image and left the negative exemplar types as background.
It did the opposite.

I have a more extensive user interface with includes text and tried a number f images with no luck

**Reply by glenn-jocher** [reply]
Thanks for confirming — based on the current [SAM 3 docs](https://docs.ultralytics.com/models/sam-3/) and [`SAM3SemanticPredictor` reference](https://docs.ultralytics.com/reference/models/sam/predict/), box exemplars are currently treated as positive visual prompts to find similar instances, so `labels=[0, 0]` should not be expected to invert the result as “negative exemplars”; if you need `all - exemplar class`, the current workaround is to segment the positive exemplar masks and subtract them in postprocessing, and if you’d like this tracked as a feature/bug please open a new issue with the sample image for visibility.

**Comment by Y-T-G** [comment]
For SAM3, you need to use `imgsz=1008`. Otherwise it will not produce good results. Negative labels should work correctly.
